### PR TITLE
Scripts for building preference datasets

### PR DESCRIPTION
**Pull request by natolambert**

### CODE REVIEW

File: scripts/data/preferences/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+* [ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_lower_10k](https://huggingface.co/datasets/ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_lower_10k)
+* [ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_mean_aspects](https://huggingface.co/datasets/ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_mean_aspects)
+* [ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_middle_10k](https://huggingface.co/datasets/ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_middle_10k)
+* [ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_overall](https://huggingface.co/datasets/ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_overall)
+* [ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_sharegpt](https://huggingface.co/datasets/ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_sharegpt)
+* [ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_top_10k](https://huggingface.co/datasets/ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_top_10k)
+* [ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_truthful_qa](https://huggingface.co/datasets/ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_truthful_qa)
+* [ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_ultrachat](https://huggingface.co/datasets/ai2-adapt-dev/tulu-2.5-prefs-ultrafeedback_ultrachat)
```
Comment (vwxyzjn) [inline scripts/data/preferences/README.md]: Love this!

File: scripts/data/preferences/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# Current preference datasets
```
Comment (vwxyzjn) [inline scripts/data/preferences/README.md]: Should we move this to `docs/datasets`? Eventually, we should move the `scripts/readme.md` there somehow, too.

File: scripts/data/preferences/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+## Chat
```
Comment (vwxyzjn) [inline scripts/data/preferences/README.md]: Could you add the commend you used to create these datasets?

File: scripts/data/preferences/hh-harmless.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+)
+
+print(f"{multiprocessing.cpu_count()=}")
+ds.push_to_hub("ai2-adapt-dev/hh-rlhf-harmless")
+api.upload_file(
+    path_or_fileobj=__file__,
+    path_in_repo="create_dataset.py",
+    repo_id="ai2-adapt-dev/hh-rlhf-harmless",
+    repo_type="dataset",
+)
```
Comment (vwxyzjn) [inline scripts/data/preferences/hh-harmless.py]: Could we make this optional? Maybe commenting out them by default, so people don't accidentally push them to hub directly. 

An alternative is to use the user's default hf entity, so they don't accidentally push to the one the people are using.

```
# debug
python -i scripts/data/preferences/hh-harmless.py --push_to_hub
# actual push
python scripts/data/preferences/hh-harmless.py --push_to_hub --hf_entity trl-internal-testing
```

**Review by vwxyzjn — COMMENTED** [review]
Thanks for doing this!

File: scripts/data/preferences/README.md (hunk 2 above)
Comment (natolambert) [reply scripts/data/preferences/README.md]: oh we have docs!

**Comment by ljvmiranda921** [comment]
I uploaded the outputs of the UF replication pipeline here: https://huggingface.co/datasets/ai2-adapt-dev/ultrafeedback-pipeline-replication 
Will just spend some time documenting these and will write a loader for this as well.

**Review by vwxyzjn — APPROVED** [review]
LGTM! Thanks for the change!

**Pull request merged**